File: Gramaticas/C/Gramatica_4/G_4.c
/*
Objetivo:
    Revisar si la cadena pertenece a L(g4) = {abb, ab}.
*/

#include <stdio.h>
#include <string.h>

int main() {
    FILE *archivo = fopen("G_4.txt", "r");
    char cadena[100];
    
    while (fgets(cadena, 100, archivo)) {
        cadena[strcspn(cadena, "\n")] = 0;
        
        if (strcmp(cadena, "abb") == 0 || strcmp(cadena, "ab") == 0) {
            printf("%s acepta\n", cadena);
        } else {
            printf("%s NO acepta\n", cadena);
        }
    }
    fclose(archivo);
    return 0;
}

File: Gramaticas/Python/Gramatica_4/G_4.py
"""
Objetivo:
    Revisar si la cadena pertenece a L(g4) = {abb, ab}.

Entrada:
    Archivo "G_4.txt".

Salida:
    "<cadena> acepta" o "<cadena> NO acepta".
"""

archivo = open("G_4.txt")

for linea in archivo:
    cadena = linea.strip()
    
    if cadena == "abb" or cadena == "ab":
        print(cadena, "acepta")
    else:
        print(cadena, "NO acepta")

archivo.close()
